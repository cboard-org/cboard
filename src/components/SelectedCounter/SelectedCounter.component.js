import React from 'react';
import PropTypes from 'prop-types';
import './SelectedCounter.css';

const propTypes = {
  count: PropTypes.number,
  text: PropTypes.string
};

const defaultProps = {
  count: 0,
  text: 'items selected'
};

const SelectedCounterComponent = ({ count, text }) => (
  <div className="SelectedCounter">
    <span>{`${count} ${text}`}</span>
  </div>
);

SelectedCounterComponent.propTypes = propTypes;
SelectedCounterComponent.defaultProps = defaultProps;

export default SelectedCounterComponent;
