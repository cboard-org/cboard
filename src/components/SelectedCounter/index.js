export { default } from './SelectedCounter.component';
