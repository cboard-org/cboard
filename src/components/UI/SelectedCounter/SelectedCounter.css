.SelectedCounter {
  align-items: center;
  display: flex;
  justify-content: center;
  padding: 0em 1em;
}
