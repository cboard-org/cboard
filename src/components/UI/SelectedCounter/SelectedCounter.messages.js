import { defineMessages } from 'react-intl';

export default defineMessages({
  items: {
    id: 'cboard.components.SelectedCounter.items',
    defaultMessage: 'items'
  }
});
