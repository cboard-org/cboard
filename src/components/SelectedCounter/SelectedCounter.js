import React from 'react';
import PropTypes from 'prop-types';
import './SelectedCounter.css';

const SelectedCounter = ({ count, text }) => (
  <div className="SelectedCounter__container">
    <span>{`${count} ${text}`}</span>
  </div>
);

SelectedCounter.propTypes = {
  count: PropTypes.number,
  text: PropTypes.string
};

SelectedCounter.defaultProps = {
  count: 0,
  text: 'items selected'
};

export default SelectedCounter;
