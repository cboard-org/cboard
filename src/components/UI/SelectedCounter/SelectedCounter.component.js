import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import messages from './SelectedCounter.messages';
import './SelectedCounter.css';

const propTypes = {
  count: PropTypes.number
};

const defaultProps = {
  count: 0
};

const SelectedCounter = ({ count }) => (
  <div className="SelectedCounter">
    <span>
      {count} <FormattedMessage {...messages.items} />
    </span>
  </div>
);

SelectedCounter.propTypes = propTypes;
SelectedCounter.defaultProps = defaultProps;

export default SelectedCounter;
