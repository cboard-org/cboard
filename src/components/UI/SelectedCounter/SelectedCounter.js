import React from 'react';
import PropTypes from 'prop-types';

import './SelectedCounter.css';

const propTypes = {
  count: PropTypes.number,
  text: PropTypes.string
};

const defaultProps = {
  count: 0,
  text: 'items selected'
};

const SelectedCounter = ({ count, text }) => (
  <div className="SelectedCounter">
    <span>{`${count} ${text}`}</span>
  </div>
);

SelectedCounter.propTypes = propTypes;
SelectedCounter.defaultProps = defaultProps;

export default SelectedCounter;
