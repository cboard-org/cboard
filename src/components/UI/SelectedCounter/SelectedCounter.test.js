import React from 'react';
import { shallowMatchSnapshot } from '../../../common/test_utils';
import SelectedCounter from './SelectedCounter.component';

describe('<SelectedCounter /> tests', () => {
  test('default renderer', () => {
    shallowMatchSnapshot(<SelectedCounter count={1} text="abcd" />);
  });
});
