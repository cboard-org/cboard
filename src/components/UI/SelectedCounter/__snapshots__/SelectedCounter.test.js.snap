// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectedCounter /> tests default renderer 1`] = `
<div
  className="SelectedCounter"
>
  1
   
  <FormattedMessage
    defaultMessage="items"
    id="cboard.components.SelectedCounter.items"
    values={Object {}}
  />
</div>
`;
